python/tabby-eval/v2-modal/eval_code_completion_jsonl.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+import logging
+
+import pandas as pd
+from Levenshtein import ratio
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(message)s',
+    datefmt='%Y-%m-%d %H:%M:%S'
+)
+
+
+def evaluation(prediction_jsonl_file):
+    df = pd.read_json(prediction_jsonl_file, lines=True)
+    total_score = 0
+
+    for index, row in df.iterrows():
+        groundtruth = row['groundtruth'].split('\n')
+        prediction = row['prediction'].split('\n')
+
+        # Get the first non-empty line
+        groundtruth_first_non_empty = next((s for s in groundtruth if s.strip()), "")
+        prediction_first_non_empty = next((s for s in prediction if s.strip()), "")
+
+        # Calculate the ratio between the two
+        score = ratio(groundtruth_first_non_empty, prediction_first_non_empty)
+
+        # Add the score to the total score
+        total_score += score
```

Comment:
we want at least following metrics:
1. line accuracy (exact match).
2. block accuracy (exact match).
3. line edit distance.
4. block edit distance.

Reply:
These metrics have been added to the calculations